src/main/java/org/asciidoc/intellij/lexer/asciidoc.flex

```diff
@@ -400,19 +400,21 @@ LINKEND = "]"
   [^]                  { return AsciiDocTokenTypes.REF; }
 }
 
-<LINKFILE> {
-  "#"                  { yybegin(LINKANCHOR); return AsciiDocTokenTypes.LINKANCHOR; }
+<LINKFILE, LINKANCHOR> {
   {LINKTEXT_START}     { yybegin(LINKTEXT); return AsciiDocTokenTypes.LINKTEXT_START; }
+}
+
+<LINKFILE> {
+  "#"                  { yybegin(LINKANCHOR); return AsciiDocTokenTypes.SEPARATOR; }
   [^]                  { return AsciiDocTokenTypes.LINKFILE; }
 }
 
 <LINKANCHOR> {
-  {LINKTEXT_START}     { yybegin(LINKTEXT); return AsciiDocTokenTypes.LINKTEXT_START; }
   [^]                  { return AsciiDocTokenTypes.LINKANCHOR; }
 }
 
 <LINKTEXT> {
-  {LINKEND}            { yybegin(YYINITIAL); return AsciiDocTokenTypes.LINKEND; }
+  {LINKEND}            { yybegin(INSIDE_LINE); return AsciiDocTokenTypes.LINKEND; }
```

Comment:
@bbrenne - Once the link text is complete, we would be again inside of a line, not at the beginning of a block. Therefore INSIDE_LINE as next state.